csm/core/services/audit_log.py

```diff
@@ -89,7 +91,10 @@ async def retrieve_by_range(self, component, limits,
         if limits and limits.limit:
             query = query.limit(limits.limit)
 
-        query = query.order_by(COMPONENT_MODEL_MAPPING[component]["field"], "desc")
+        if sort:
```

Comment:
No need of IF here. You are already setting the default value of SORT before passing it to this function.

Reply:
Hi @pawankumarsrivastava , sort is a optional parameter for retrieve_by_range function. It gets set to default values if the caller is  AuditLogShowView controller. But that is not the case if caller is AuditLogDownloadView controller. That is why, IF checking is required there.

Reply:
Sort is option but in case of sort missing you have already defined the default values. 

sort_by = fields.Str(data_key='sortby', missing="timestamp", default="timestamp")
direction = fields.Str(data_key='dir', validate=validate.OneOf(['desc', 'asc']),
        missing='desc', default='desc')

Do you understand why have you added this code?

Reply:
@rohitkolapkar Add TODO.